src/main/java/domain/LadderGame.java

```diff
@@ -0,0 +1,62 @@
+package domain;
+
+import domain.ladder.Ladder;
+import domain.ladder.Line;
+import domain.player.Player;
+import domain.player.Players;
+import domain.prize.Prizes;
+import domain.prize.Results;
+
+public class LadderGame {
+
+    private final Ladder ladder;
+    private final Players players;
+    private final Prizes prizes;
+    private final Results results;
+
+
+    public LadderGame(Ladder ladder, Players players, Prizes prizes) {
+        this.ladder = ladder;
+        this.players = players;
+        this.prizes = prizes;
+        this.results = new Results();
+    }
+
+    public Results run() {
+        if (isDone()) {
+            return results;
+        }
+        play();
+        finish();
+        return results;
+    }
+
+    private boolean isDone() {
+        return results.isSameSizeAs(players.getPlayerSize());
+    }
+
+    private void finish() {
+        for (final Player player : players.getPlayers()) {
+            int position = player.getPosition();
+            results.addResult(player, prizes.getOnePrizeByIndex(position));
+        }
+    }
+
+    private void play() {
+        for (final Line line : ladder.getLadder()) {
+            players.move(line);
+        }
```

Comment:
```suggestion
ladder.getLadder()
                .forEach(players::move);
```

당연히 아시겠지만 요론 방법도 있습니다-! 사실 저는 for문을 더 선호하긴 합니다-!